op-service/txmgr/txmgr.go

```diff
@@ -74,12 +87,55 @@ type SimpleTxManager struct {
 	Config // embed the config directly
 	name   string
 
-	backend ReceiptSource
+	backend L1Connection
```

Comment:
Why is this an L1 connection? It could be used to send txs to L2? It really should be a generic backend?

Reply:
You're right, it could be a generic "eth backend". I just extended the name because it is no longer just receipts.

Reply:
types are found here: https://github.com/ethereum/go-ethereum/blob/d0a4989a8def7e6bad182d1513e8d4a093c1672d/accounts/abi/bind/backend.go

Reply:
nice. We actually only need a subset of that so I'll probably rename `L1Connection` rather than import that set.